docs/features/software-templates/writing-custom-filters.md

```diff
@@ -0,0 +1,145 @@
+---
+id: writing-custom-filters
+title: Writing Custom Filters
+description: How to write your own filters for software templates
+---
+
+By default, the scaffolder page provides a set of filters that you can use to filter the available software templates. However, you can also write your own filters based on custom kinds and properties, and mix and match them with the default filters.
+
+## Writing a Custom Filter
+
+Writing a custom filter for the scaffolder is almost identical to [writing custom filters for the catalog](https://backstage.io/docs/features/software-catalog/catalog-customization/#customize-filters).
```

Comment:
Hi @awanlin, I thought that perhaps users would expect filter customization to appear under the plugin docs for each plugin. Users working on the scaffolder specifically might have a hard time finding what they need under `software-catalog > catalog-customization`.

Should I extend the original `catalog-customization` to show users how to add custom filters for existing types (custom Tags picker) and keep the other one for new types (SecurityTier picker)? I can then remove the redundant examples in `writing-custom-filters.md` and leave only the link to `catalog-customization` page.

Reply:
The only documentation here should be how to pass the filter to be used by the Scaffolder. This should simply point to the Catalog filter documentation. 

Ideally would like to avoid multiple examples for the same concept. We could probably just add a sentence to the Catalog filters explaining that this can be used to also provide custom filters for existing filters if they have alternative implementations they would like.

Reply:
Thanks for the feedback @awanlin!

As per the recent changes requested by @benjdlambert, customizing Scaffolder filters requires passing them as children components to the `Route`, as explained in a comment below. Would this be enough reason to keep a separate documentation file and example? I'm not sure if those changes are good to go, but I'll be able to fix up the docs once I get some feedback. I'm aware we're going for consistency/simplicity here, just not sure how to reconcile both the requested changes! 😅